File: app/src/main/java/com/roman/batsu/utils/network/NetworkChecker.java
package com.roman.batsu.utils.network;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkChecker {




    private static NetworkInfo getNetworkInfo(Context context){
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        return cm.getActiveNetworkInfo();
    }

    public static boolean isNetworkAvailable(Context context){
        NetworkInfo info = NetworkChecker.getNetworkInfo(context);
        return (info != null && info.isConnected());
    }





}

File: app/src/main/java/com/roman/batsu/utils/network/NetworkChecker.kt
package com.roman.batsu.utils.network

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo
import androidx.appcompat.app.AppCompatActivity

object NetworkChecker {


    @JvmStatic
    fun isNetworkAvailable(context: Context?): Boolean {
        val connectivityManager=context?.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val networkInfo=connectivityManager.activeNetworkInfo
        return  networkInfo!=null && networkInfo.isConnected
    }

}